Rebus.Idempotency/MessageId.cs

```diff
@@ -0,0 +1,58 @@
+namespace Rebus.Idempotency
+{
+    public class MessageId
+    {
+        public string OriginalMessageId;
+        public int DeferCount = 0;
+
+        public MessageId(string messageId)
+        {
+            OriginalMessageId = messageId;
+        }
+
+        public MessageId(string messageId, int? deferCount)
```

Comment:
It's a bit strange to have one constructor with just the `string` and another one with a nullable parameter. Can't there be just one constructor `MessageId(string messageId, int deferCount = 0)`?